### PR TITLE
Switch to go mod

### PR DESCRIPTION
**Pull request by jackzampolin**
This PR switches dependency management to using `go mod`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,38 +1,38 @@
 GOTOOLS = \
 	github.com/mitchellh/gox \
-	github.com/golang/dep/cmd/dep \
 	github.com/golangci/golangci-lint/cmd/golangci-lint \
 	github.com/gogo/protobuf/protoc-gen-gogo \
 	github.com/square/certstrap
 GOBIN?=${GOPATH}/bin
 PACKAGES=$(shell go list ./...)
 OUTPUT?=build/tendermint
+GO111MODULE=on
```
Comment (alessio) [inline Makefile]:
```suggestion
export GO111MODULE = on
```

Comment (jackzampolin) [reply Makefile]: done

Comment (alessio) [reply Makefile]: That is required to be propagated to all commands - `Make` != `bash`

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: 2
 defaults: &defaults
   working_directory: /go/src/github.com/tendermint/tendermint
   docker:
-    - image: circleci/golang
+    - image: circleci/golang:1.12
```
Comment (Liamsi) [inline .circleci/config.yml]: That's not necessary. circleci/golang points to the latest go version.

File: .circleci/config.yml [hunk 3]
```diff
@@ -278,15 +267,11 @@ jobs:
...
-            make get_tools get_vendor_deps
+            make get_tools
       - persist_to_workspace:
           root: .
           paths:
             - "release-version.source"
-      - save_cache:
-          key: v1-release-deps-{{ .Branch }}-{{ .Revision }}
-          paths:
-            - "vendor"
```
Comment (Liamsi) [inline .circleci/config.yml]: Please, do not delete the parts that enable a cache (but make it use modules too)!  

There is still a 
```yml
- restore_cache:
          keys:
            - v1-release-deps-{{ .Branch }}-{{ .Revision }}
```
left below. If there are any good reasons to not use a cache, then we'd need to delete those lines too.

**Comment by jackzampolin** [comment]
Any way we could do the rest in a follow up PR. I'm not super firmiliar with where `go mod` caches the deps... maybe @alessio could help out here?

**Comment by Liamsi** [comment]
We should remove both `save_cache` and `restore_cache` (and not only the first part). Everything else can be done in a follow-up PR. 
Regarding:
> 'm not super firmiliar with where go mod caches the deps... maybe @alessio could help out here?

You can have a look here: https://github.com/tendermint/go-amino/blob/fc5dc13311ae690b06c40efb18379976f962c6ce/.circleci/config.yml#L9-L22

I can take this over in a follow-up PR, too.

**Comment by jackzampolin** [comment]
If I pull that out a number of circle jobs fail. Not 100% sure how that works @Liamsi cc @mircea-c

**Comment by Liamsi** [comment]
As commented above only the cache with the key `v1-release-deps-{{ .Branch }}-{{ .Revision }}` should have been deleted. Not *all* caches. This is easy to fix by reverting the last commit and removing all remaining `restore_cache` calls with that particular key. We can (optionally) introduce a separate cache (with a new key, e.g. `go-mod-v1-{{ checksum "go.sum" }}`) for the modules under `$GOPATH/pkg/mod`. Optionally, because (AFAIU) they are already covered by the `v3-pkg-cache` (it caches the whole `$GOPATH/pkg/` including `$GOPATH/pkg/mod`). Maybe we can just keep that one as it is and bump the version to indicate the change (additionally cache `mod`). Also not sure about best practices. Having a separate key by concern would definitely increase the readability and lead to a clearer separation of concerns. I think we should go with the simple fix (keep `v3-pkg-cache` and bump it to `v4-pkg-cache`) outlined above and not care for the latter in this PR (but if someone has the bandwidth we can do this in a follow-up PR).

**Comment by Liamsi** [comment]
See: https://github.com/tendermint/tendermint/pull/3623

Independent from that, these need to be fixed too:
 - [ ] there are still traces of `dep ensure` etc. in documentation ...
 - [ ] decide: we could make this a non-breaking change by keeping support for both: dep and go-modules (at least for a while); probably ask our community for feedback on that!
 - [ ] Gopkg.toml and Gopk.lock are still part of the repo but do not make sense without maintaining them too

File: .circleci/config.yml [hunk 4]
```diff
@@ -157,7 +146,7 @@ jobs:
             for pkg in $(go list github.com/tendermint/tendermint/... | circleci tests split --split-by=timings); do
               id=$(basename "$pkg")
 
-              go test -v -timeout 5m -race -coverprofile=/tmp/workspace/profiles/$id.out -covermode=atomic "$pkg" | tee "/tmp/logs/$id-$RANDOM.log"
+              GO111MODULE=on go test -v -timeout 5m -mod=readonly -race -coverprofile=/tmp/workspace/profiles/$id.out -covermode=atomic "$pkg" | tee "/tmp/logs/$id-$RANDOM.log"
```
Comment (melekes) [inline .circleci/config.yml]: Won't it be better to set it in top-level defaults?

```
defaults: &defaults
    environment:
      GO111MODULE: on
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -42,8 +42,7 @@ dependencies so we can get away with telling people they can just `go get` our
 software.
 
 Since some dependencies are not under our control, a third party may break our
-build, in which case we can fall back on `dep ensure` (or `make
-get_vendor_deps`). Even for dependencies under our control, dep helps us to
+build, in which case we can fall back on `go mod tidy`. Even for dependencies under our control, dep helps us to
```
Comment (melekes) [inline CONTRIBUTING.md]: Even for dependencies under our control, dep helps us to

=> 

Even for dependencies under our control, go mod helps us to

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,51 @@
+module github.com/tendermint/tendermint
+
+go 1.12
+
+require (
+	github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
```
Comment (melekes) [inline go.mod]: can we structure this like Gopkg.toml?

```
# Allow only patch releases for serialization libraries
# Allow only minor releases for other libraries
## Test libraries
## XXX
## YYY
```

**Comment by Liamsi** [comment]
> Should we also remove Gopkg.toml and Gopkg.lock files?

As this PR currently removes `dep` completely, they won't be maintained and they should be removed. It might have been nice to keep support for both (dep and modules) but this is probably not worth the effort.

**Comment by ethanfrey** [comment]
I like this.

We recently migrated to go mod as well, but had to drop support for everything less than 1.11.4 (early 1.11 had some issues with go mod). If you are going to break golang version support, breaking dep support at the same time is fine in my opinion.

Maybe good to have a stable release (0.31.5 is it?) that supports the 1.10 and dep, then just make a breaking change to build but not many features to allow an easy upgrade path for project (just the build system and not the apis)

**Comment by Liamsi** [comment]
> Maybe good to have a stable release (0.31.5 is it?) that supports the 1.10 and dep

We are already expecting go1.11.4 or higher for a while now: https://github.com/tendermint/tendermint#minimum-requirements

**Comment by Liamsi** [comment]
On another note. Can this file be deleted: 
https://github.com/tendermint/tendermint/blob/a076b4820219313c7b388df10c8664042d8711dc/DOCKER/Dockerfile.abci#L6-L23

File: go.mod (hunk 6 above)
Comment (Liamsi) [reply go.mod]: Related note to myself: double check that all dependencies from the latest develop survived the mode to go modules (e.g. can't find "github.com/etcd-io/bbolt" which was introduced after this PR was opened)

Comment (Liamsi) [reply go.mod]: @melekes looks like part of the tooling will rewrite the go.mod file to be alphabetically sorted...

File: Makefile [hunk 7]
```diff
@@ -1,38 +1,39 @@
...
 
-check: check_tools get_vendor_deps
+check: check_tools
 
 ########################################
 ### Build Tendermint
 
 build:
-	CGO_ENABLED=0 go build $(BUILD_FLAGS) -tags $(BUILD_TAGS) -o $(OUTPUT) ./cmd/tendermint/
+	CGO_ENABLED=0 go build -mod=readonly $(BUILD_FLAGS) -tags $(BUILD_TAGS) -o $(OUTPUT) ./cmd/tendermint/
```
Comment (melekes) [inline Makefile]: can we add `-mod=readonly` to `$(BUILD_FLAGS)`?

File: go.mod [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+	github.com/btcsuite/btcd v0.0.0-20190115013929-ed77733ec07d
```
Comment (melekes) [inline go.mod]: looks like this got updated!

File: go.mod [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+	github.com/spf13/cast v1.3.0
+	github.com/spf13/cobra v0.0.1
+	github.com/spf13/jwalterweatherman v1.0.0
+	github.com/spf13/pflag v1.0.3
+	github.com/spf13/viper v1.0.0
+	github.com/stretchr/testify v1.2.2
+	github.com/syndtr/goleveldb v0.0.0-20181012014443-6b91fda63f2e
+	github.com/tendermint/go-amino v0.14.1
+	golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25
+	golang.org/x/net v0.0.0-20180906233101-161cd47e91fd
```
Comment (melekes) [inline go.mod]: looks like this got updated!

File: go.mod [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+	golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a
```
Comment (melekes) [inline go.mod]: looks like this got updated!

File: Makefile [hunk 11]
```diff
@@ -11,7 +11,7 @@ export GO111MODULE = on
 
 INCLUDE = -I=. -I=${GOPATH}/src -I=${GOPATH}/src/github.com/gogo/protobuf/protobuf
 BUILD_TAGS?='tendermint'
-BUILD_FLAGS = -ldflags "-X github.com/tendermint/tendermint/version.GitCommit=`git rev-parse --short=8 HEAD`"
+BUILD_FLAGS = -mod=readonly -ldflags "-X github.com/tendermint/tendermint/version.GitCommit=`git rev-parse --short=8 HEAD`"
```
Comment (Liamsi) [inline Makefile]: 👍

**Comment by melekes** [comment]
https://github.com/btcsuite/btcutil

```
git diff d4cc87b86~ ab6388e0c60a --stat -- ':!*.pb.go' ':!test'
 gcs/builder/builder.go         | 220 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------------
 gcs/builder/builder_test.go    |  37 ++++++-----------------
 gcs/gcs.go                     |  60 ++++++++++++-------------------------
 gcs/gcs_test.go                |  48 +++++-------------------------
 gcs/gcsbench_test.go           |  20 ++++++++-----
 hdkeychain/extendedkey_test.go |   2 +-
 6 files changed, 142 insertions(+), 245 deletions(-)
```

Looks like the most important change is https://github.com/btcsuite/btcutil/commit/0ecd90b8d66e9dbac3e7a56faf6d6475b26ecac4 lowering `DefaultP` const.

**Comment by melekes** [comment]
https://github.com/golang/net

```
» git diff 292b43bbf7cb8d35ddf40f8d5100ef3837cced3f~ 161cd47e91fd --stat -- ':!*.pb.go' ':!test'                                         141 ↵
 dns/dnsmessage/message.go         |    65 +-
 dns/dnsmessage/message_test.go    |    30 +
 html/parse.go                     |    99 +-
 html/parse_test.go                |     1 +
 html/testdata/webkit/template.dat |   223 +-
 http2/configure_transport.go      |     8 +-
 http2/frame.go                    |    63 +-
 http2/frame_test.go               |    44 +
 http2/go111.go                    |     9 +
 http2/h2c/h2c.go                  |   492 ++++
 http2/h2c/h2c_test.go             |    58 +
 http2/h2demo/h2demo.go            |     3 +
 http2/headermap.go                |    20 +-
 http2/hpack/hpack_test.go         |    21 +
 http2/hpack/huffman.go            |    20 +-
 http2/http2.go                    |    17 +-
 http2/not_go111.go                |     6 +
 http2/server.go                   |    30 +-
 http2/server_test.go              |    87 +-
 http2/transport.go                |    87 +-
 http2/transport_test.go           |   137 +
 icmp/endpoint.go                  |     5 +-
 icmp/extension.go                 |     6 +-
 icmp/ipv4.go                      |    12 +-
 icmp/message.go                   |     8 +-
 icmp/messagebody.go               |     6 +-
 ipv4/doc.go                       |     2 +-
 ipv4/header.go                    |    12 +
 ipv6/doc.go                       |     2 +-
 publicsuffix/list.go              |    39 +-
 publicsuffix/table.go             | 18753 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++----------------------------------------------------------------
 publicsuffix/table_test.go        |    26 +-
 trace/trace.go                    |     8 +
 33 files changed, 10795 insertions(+), 9604 deletions(-)
```

changes http2 and html packages

**Comment by melekes** [comment]
https://github.com/golang/sys

```
» git diff 7e31e0c00fa05cb5fbf4347b585621d6709e19a4~ d0b11bdaac8a --stat -- ':!*.pb.go' ':!test'                                         141 ↵
 cpu/byteorder.go                                |   30 +
 cpu/cpu.go                                      |   51 ++
 cpu/cpu_arm.go                                  |    2 +
 cpu/cpu_linux.go                                |   55 ++
 cpu/cpu_linux_arm64.go                          |   67 ++
 cpu/cpu_mips64x.go                              |    2 +
 cpu/cpu_mipsx.go                                |    2 +
 cpu/{cpu_arm64.go => cpu_other_arm64.go}        |    6 +-
 cpu/cpu_ppc64x.go                               |   23 +
 cpu/cpu_s390x.go                                |    2 +
 cpu/cpu_test.go                                 |   26 +
 cpu/cpu_x86.go                                  |    2 +
 plan9/mksyscall.go                              |  393 +++++++++++
 plan9/mksyscall.pl                              |  319 ---------
 plan9/zsyscall_plan9_386.go                     |   16 +-
 plan9/zsyscall_plan9_amd64.go                   |   16 +-
 plan9/zsyscall_plan9_arm.go                     |    4 +-
 unix/README.md                                  |   20 +-
 unix/asm_freebsd_arm64.s                        |   29 +
 unix/asm_linux_ppc64x.s                         |   12 -
 unix/asm_netbsd_arm64.s                         |   29 +
 unix/darwin_test.go                             |  210 ++++++
 unix/example_exec_test.go                       |   19 +
 unix/{example_test.go => example_flock_test.go} |    7 +-
 unix/fcntl.go                                   |    2 +-
 unix/fcntl_darwin.go                            |   18 +
 unix/linux/Dockerfile                           |   14 +-
 unix/linux/mkall.go                             |   55 +-
 unix/linux/mksysnum.go                          |  143 ++++
 unix/linux/mksysnum.pl                          |   85 ---
 unix/linux/types.go                             |   46 +-
 unix/mkall.sh                                   |   78 ++-
 unix/mkasm_darwin.go                            |   61 ++
 unix/mkerrors.sh                                |   39 +-
 unix/mkpost.go                                  |   14 +-
 unix/mksyscall.go                               |  402 +++++++++++
 unix/mksyscall.pl                               |  341 ----------
 unix/mksyscall_aix_ppc.go                       |  404 +++++++++++
 unix/mksyscall_aix_ppc.pl                       |  384 -----------
 unix/mksyscall_aix_ppc64.go                     |  602 +++++++++++++++++
 unix/mksyscall_aix_ppc64.pl                     |  579 ----------------
 unix/mksyscall_solaris.go                       |  335 ++++++++++
 unix/mksyscall_solaris.pl                       |  294 --------
 unix/mksysnum.go                                |  190 ++++++
 unix/mksysnum_darwin.pl                         |   39 --
 unix/mksysnum_dragonfly.pl                      |   50 --
 unix/mksysnum_freebsd.pl                        |   50 --
 unix/mksysnum_netbsd.pl                         |   58 --
 unix/mksysnum_openbsd.pl                        |   50 --
 unix/sendfile_test.go                           |   98 +++
 unix/sockcmsg_unix.go                           |   25 +-
 unix/syscall_aix.go                             |   18 +-
 unix/syscall_darwin.go                          |   63 +-
 unix/syscall_darwin_386.go                      |   23 +-
 unix/syscall_darwin_amd64.go                    |   23 +-
 unix/syscall_darwin_arm.go                      |   26 +-
 unix/syscall_darwin_arm64.go                    |   26 +-
 unix/syscall_darwin_libSystem.go                |   31 +
 unix/syscall_darwin_test.go                     |   44 ++
 unix/syscall_dragonfly.go                       |   15 +
 unix/syscall_freebsd.go                         |    7 +
 unix/syscall_freebsd_arm64.go                   |   52 ++
 unix/syscall_linux.go                           |  111 +++-
 unix/syscall_linux_386.go                       |    1 +
 unix/syscall_linux_amd64.go                     |   19 +-
 unix/syscall_linux_arm.go                       |    9 +
 unix/syscall_linux_arm64.go                     |    1 +
 unix/syscall_linux_mips64x.go                   |   10 +-
 unix/syscall_linux_mipsx.go                     |    1 +
 unix/syscall_linux_ppc64x.go                    |    1 +
 unix/syscall_linux_riscv64.go                   |    4 +
 unix/syscall_linux_s390x.go                     |    1 +
 unix/syscall_linux_sparc64.go                   |    1 +
 unix/syscall_linux_test.go                      |   48 ++
 unix/syscall_netbsd.go                          |   15 +
 unix/syscall_netbsd_arm64.go                    |   33 +
 unix/syscall_netbsd_test.go                     |   16 +
 unix/syscall_openbsd.go                         |    7 +
 unix/syscall_solaris.go                         |    7 +
 unix/syscall_unix.go                            |   15 -
 unix/syscall_unix_gc.go                         |    2 +-
 unix/syscall_unix_gc_ppc64x.go                  |   24 +
 unix/syscall_unix_test.go                       |   46 +-
 unix/types_freebsd.go                           |    2 +-
 unix/types_netbsd.go                            |    2 +
 unix/xattr_bsd.go                               |   15 +-
 unix/xattr_test.go                              |    8 +-
 unix/zerrors_freebsd_arm64.go                   | 1794 +++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zerrors_linux_386.go                       |   84 ++-
 unix/zerrors_linux_amd64.go                     |   84 ++-
 unix/zerrors_linux_arm.go                       |   84 ++-
 unix/zerrors_linux_arm64.go                     |   84 ++-
 unix/zerrors_linux_mips.go                      |   84 ++-
 unix/zerrors_linux_mips64.go                    |   84 ++-
 unix/zerrors_linux_mips64le.go                  |   84 ++-
 unix/zerrors_linux_mipsle.go                    |   84 ++-
 unix/zerrors_linux_ppc64.go                     |   86 ++-
 unix/zerrors_linux_ppc64le.go                   |   86 ++-
 unix/zerrors_linux_riscv64.go                   |   84 ++-
 unix/zerrors_linux_s390x.go                     |   84 ++-
 unix/zerrors_linux_sparc64.go                   | 4240 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-------------------------------------------------
 unix/zerrors_netbsd_arm64.go                    | 1762 ++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_aix_ppc.go                        |    2 +-
 unix/zsyscall_aix_ppc64.go                      |    2 +-
 unix/zsyscall_aix_ppc64_gc.go                   |    2 +-
 unix/zsyscall_aix_ppc64_gccgo.go                |    2 +-
 unix/zsyscall_darwin_386.1_11.go                | 1810 ++++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_darwin_386.go                     | 1174 ++++++++++++++++++++++++++------
 unix/zsyscall_darwin_386.s                      |  284 ++++++++
 unix/zsyscall_darwin_amd64.1_11.go              | 1810 ++++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_darwin_amd64.go                   | 1189 +++++++++++++++++++++++++++------
 unix/zsyscall_darwin_amd64.s                    |  286 ++++++++
 unix/zsyscall_darwin_arm.1_11.go                | 1793 +++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_darwin_arm.go                     | 1152 ++++++++++++++++++++++++++------
 unix/zsyscall_darwin_arm.s                      |  282 ++++++++
 unix/zsyscall_darwin_arm64.1_11.go              | 1793 +++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_darwin_arm64.go                   | 1152 ++++++++++++++++++++++++++------
 unix/zsyscall_darwin_arm64.s                    |  282 ++++++++
 unix/zsyscall_dragonfly_amd64.go                |  137 +++-
 unix/zsyscall_freebsd_386.go                    |    2 +-
 unix/zsyscall_freebsd_amd64.go                  |    2 +-
 unix/zsyscall_freebsd_arm.go                    |    2 +-
 unix/zsyscall_freebsd_arm64.go                  | 2015 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_linux_386.go                      |   59 +-
 unix/zsyscall_linux_amd64.go                    |   76 +--
 unix/zsyscall_linux_arm.go                      |   69 +-
 unix/zsyscall_linux_arm64.go                    |   59 +-
 unix/zsyscall_linux_mips.go                     |   59 +-
 unix/zsyscall_linux_mips64.go                   |   89 ++-
 unix/zsyscall_linux_mips64le.go                 |   89 ++-
 unix/zsyscall_linux_mipsle.go                   |   59 +-
 unix/zsyscall_linux_ppc64.go                    |   59 +-
 unix/zsyscall_linux_ppc64le.go                  |   59 +-
 unix/zsyscall_linux_riscv64.go                  |   39 +-
 unix/zsyscall_linux_s390x.go                    |   59 +-
 unix/zsyscall_linux_sparc64.go                  |  225 ++++++-
 unix/zsyscall_netbsd_386.go                     |    2 +-
 unix/zsyscall_netbsd_amd64.go                   |    2 +-
 unix/zsyscall_netbsd_arm.go                     |    2 +-
 unix/zsyscall_netbsd_arm64.go                   | 1826 ++++++++++++++++++++++++++++++++++++++++++++++++++
 unix/zsyscall_openbsd_386.go                    |    2 +-
 unix/zsyscall_openbsd_amd64.go                  |    2 +-
 unix/zsyscall_openbsd_arm.go                    |    2 +-
 unix/zsyscall_solaris_amd64.go                  |    2 +-
 unix/zsysnum_darwin_386.go                      |    2 +-
 unix/zsysnum_darwin_amd64.go                    |    6 +-
 unix/zsysnum_darwin_arm.go                      |    2 +-
 unix/zsysnum_darwin_arm64.go                    |    2 +-
 unix/zsysnum_dragonfly_amd64.go                 |  176 ++---
 unix/zsysnum_freebsd_386.go                     |  452 ++++++-------
 unix/zsysnum_freebsd_amd64.go                   |  452 ++++++-------
 unix/zsysnum_freebsd_arm.go                     |  452 ++++++-------
 cpu/byteorder.go                                |   30 +
 cpu/cpu.go                                      |   51 ++
 cpu/cpu_arm.go                                  |    2 +
 cpu/cpu_linux.go                                |   55 ++
 cpu/cpu_linux_arm64.go                          |   67 ++
 cpu/cpu_mips64x.go                              |    2 +
 cpu/cpu_mipsx.go                                |    2 +
 cpu/{cpu_arm64.go => cpu_other_arm64.go}        |    6 +-
 cpu/cpu_ppc64x.go                               |   23 +
 cpu/cpu_s390x.go                                |    2 +
 cpu/cpu_test.go                                 |   26 +
 cpu/cpu_x86.go                                  |    2 +
 plan9/mksyscall.go                              |  393 +++++++++++
 plan9/mksyscall.pl                              |  319 ---------
 plan9/zsyscall_plan9_386.go                     |   16 +-
 plan9/zsyscall_plan9_amd64.go                   |   16 +-
 plan9/zsyscall_plan9_arm.go                     |    4 +-
 unix/README.md                                  |   20 +-
 unix/asm_freebsd_arm64.s                        |   29 +
 unix/asm_linux_ppc64x.s                         |   12 -
 unix/asm_netbsd_arm64.s                         |   29 +
 unix/darwin_test.go                             |  210 ++++++
 unix/example_exec_test.go                       |   19 +
 unix/{example_test.go => example_flock_test.go} |    7 +-
 unix/fcntl.go                                   |    2 +-
 unix/fcntl_darwin.go                            |   18 +
 unix/linux/Dockerfile                           |   14 +-
 unix/linux/mkall.go                             |   55 +-
 unix/linux/mksysnum.go                          |  143 ++++
 unix/linux/mksysnum.pl                          |   85 ---
 unix/linux/types.go                             |   46 +-
 unix/mkall.sh                                   |   78 ++-
 unix/mkasm_darwin.go                            |   61 ++
 unix/mkerrors.sh                                |   39 +-
 unix/mkpost.go                                  |   14 +-
 unix/mksyscall.go                               |  402 +++++++++++
 unix/mksyscall.pl                               |  341 ----------
 unix/mksyscall_aix_ppc.go                       |  404 +++++++++++
 unix/mksyscall_aix_ppc.pl                       |  384 -----------
 unix/mksyscall_aix_ppc64.go                     |  602 +++++++++++++++++
 unix/mksyscall_aix_ppc64.pl                     |  579 ----------------
 unix/mksyscall_solaris.go                       |  335 ++++++++++
 unix/mksyscall_solaris.pl                       |  294 --------
 unix/mksysnum.go                                |  190 ++++++
 unix/mksysnum_darwin.pl                         |   39 --
 unix/mksysnum_dragonfly.pl                      |   50 --
 unix/mksysnum_freebsd.pl                        |   50 --
 unix/mksysnum_netbsd.pl                         |   58 --
```

**Pull request merged**